# THỐNG KÊ MÁY TÍNH - THƯỜNG KỲ 2

Họ và tên: Đoàn Văn Quý Tỷ

MSSV: 22704831

## Bài 1:(4 điểm) Một công ty sản xuất thiết bị điện tử muốn đánh giá tỉ lệ sản phẩm lỗi trong quy trình sản xuất mới. Họ đã tiến hành kiểm tra ngẫu nhiên 500 thiết bị và phát hiện 35 thiết bị có lỗi.

```python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
```

### a) Tính tỉ lệ thiết bị lỗi trong mẫu và ước lượng độ lệch chuẩn của tỉ lệ mẫu. (1 điểm)

In [13]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm
import math
from scipy import stats
import pandas as pd
# from scipy.st
# ats import t
import statsmodels.api as sm
from sklearn import linear_model
import statsmodels.formula.api as smf

### b) Xây dựng khoảng ước lượng cho tỉ lệ thiết bị lỗi trong toàn bộ dây chuyền sản xuất với độ tin cậy 95%. (2 điểm)

### c) Tiêu chuẩn của ngành yêu cầu tỉ lệ sản phẩm lỗi không được vượt quá 8%. Dựa vào khoảng ước lượng mà bạn đã tính được, hãy cho biết liệu quy trình sản xuất mới có đáp ứng được tiêu chuẩn của ngành hay không? (1 điểm)

## Bài 2: 
Một nhà sản xuất pin mặt trời tuyên bố rằng tuổi thọ trung bình của sản phẩm pin mặt trời mới của họ là ít nhất 15 năm. Một viện nghiên cứu độc lập muốn kiểm tra tuyên bố này và đã theo dõi 18 pin mặt trời cho đến khi chúng hết tuổi thọ sử dụng. Dữ liệu về tuổi thọ (tính bằng năm) được ghi lại như sau:

```python
solar_panel_lifetime = np.array([
    14.2, 15.7, 13.8, 16.2, 14.5, 15.9, 13.5, 14.8, 16.5,
    15.1, 14.7, 15.4, 13.9, 15.8, 14.4, 15.3, 14.9, 15.5
])
```

### a) Với mức ý nghĩa α = 0.05, hãy thực hiện kiểm định để xác định liệu tuổi thọ trung bình của pin mặt trời có thực sự đạt được ít nhất 15 năm như nhà sản xuất tuyên bố hay không? (3 điểm)

B1. Thiết lập giả thuyết và kiểm tra điều kiện

𝐻0:𝜇<=15(không vượt quá tuổi thọ trung bình)
 

𝐻1:𝜇>15(vượt quá tuổi thọ trung bình)
 

Điều kiện kiểm định: Mẫu nhỏ ( 𝑛=18
 ) nhưng không biết  𝜎
 , theo định lý giới hạn trung tâm,  𝑥¯
  xấp xỉ phân phối chuẩn -> dùng phân phối t phía trái

B2. Mức ý nghĩa kiểm định

alpha = 0.05

B3. Tính giá trị kiểm định từ mẫu
$ t = \frac{\bar{x} - \mu_0}{s / \sqrt{n}} $ = 0.0284

B4. Xác định  p-value
p-value:


$$
\text{p-value} \approx 0 > \alpha = 0.05 \Rightarrow \text{Không bác bỏ } H_0
$$

B5. Kết luận
- p-value **lớn** mức ý nghĩa
- **Kết luận:** tuổi thọ trung bình của pin mặt trời có thực sự đạt được ít nhất 15 năm

In [18]:
solar_panel_lifetime = np.array([
    14.2, 15.7, 13.8, 16.2, 14.5, 15.9, 13.5, 14.8, 16.5,
    15.1, 14.7, 15.4, 13.9, 15.8, 14.4, 15.3, 14.9, 15.5
])

In [21]:
# Dữ liệu
mu_0 = 15
x_bar = solar_panel_lifetime.mean()
sigma = solar_panel_lifetime.std()
n = len(solar_panel_lifetime)
alpha = 0.05

# Trị kiểm định
t = (x_bar - mu_0) / (sigma / np.sqrt(n))
print("Trị kiểm định t =", round(t, 4))

t_crit = stats.t.ppf(1 - alpha, df=n-1)
print(f"Trị tới hạn: {t_crit:.4f}")

t_score = stats.t.ppf(1 - alpha, df=n-1)
se = sigma / np.sqrt(n)
ci = (x_bar - t_score * se, x_bar + t_score * se)
print(f"CI 95%: {ci}")

# p-value
p_value = 1 - stats.t.cdf(t, df=n-1)
print("p-value =", round(p_value, 4))

if p_value < alpha:
    print("Kết luận theo p-value: Bác bỏ H0")
else:
    print("Kết luận theo p-value: Không bác bỏ H0")
print(x_bar)

Trị kiểm định t = 0.0284
Trị tới hạn: 1.7396
CI 95%: (14.664830216925882, 15.346280894185231)
p-value = 0.4889
Kết luận theo p-value: Không bác bỏ H0
15.005555555555556


### b) Từ kết quả trên, hãy đưa ra kết luận về tuyên bố của nhà sản xuất. Nếu bạn là chuyên gia tư vấn cho nhà sản xuất, bạn sẽ đề xuất những biện pháp gì? (1 điểm)

## Bài 3: (2 điểm)
Một nhà nghiên cứu môi trường muốn khảo sát mối quan hệ giữa mức tiêu thụ năng lượng hàng tháng (kWh) và diện tích (m²) của các căn hộ trong một khu chung cư. Sau khi thu thập dữ liệu từ 12 căn hộ, nhà nghiên cứu ghi nhận được:

```python
apartment_data = pd.DataFrame({
    'Dien_tich': [45, 60, 75, 50, 85, 95, 65, 70, 80, 55, 90, 40],
    'Tieu_thu_nang_luong': [280, 350, 420, 300, 480, 520, 380, 400, 450, 320, 500, 260]
})

print(apartment_data)

### a) Tính hệ số tương quan Pearson giữa diện tích và mức tiêu thụ năng lượng. Xây dựng mô hình hồi quy tuyến tính đơn để dự đoán mức tiêu thụ năng lượng dựa vào diện tích căn hộ. Viết phương trình hồi quy. (1 điểm)


In [10]:
apartment_data = pd.DataFrame({
    'Dien_tich': [45, 60, 75, 50, 85, 95, 65, 70, 80, 55, 90, 40],
    'Tieu_thu_nang_luong': [280, 350, 420, 300, 480, 520, 380, 400, 450, 320, 500, 260]
})

print(apartment_data)

    Dien_tich  Tieu_thu_nang_luong
0          45                  280
1          60                  350
2          75                  420
3          50                  300
4          85                  480
5          95                  520
6          65                  380
7          70                  400
8          80                  450
9          55                  320
10         90                  500
11         40                  260


In [14]:
model = sm.formula.ols('Tieu_thu_nang_luong ~ Dien_tich', data=apartment_data).fit()
print(model.summary())
print(f"Mô hình hồi quy: y = {model.params[0]:.4f} + {model.params[1]:.4f}x")
r = np.corrcoef(apartment_data["Dien_tich"], apartment_data["Tieu_thu_nang_luong"])[0, 1]
print("Hệ số tương quan:", r)


                             OLS Regression Results                            
Dep. Variable:     Tieu_thu_nang_luong   R-squared:                       0.998
Model:                             OLS   Adj. R-squared:                  0.998
Method:                  Least Squares   F-statistic:                     4456.
Date:                 Tue, 15 Apr 2025   Prob (F-statistic):           1.39e-14
Time:                         18:40:51   Log-Likelihood:                -33.636
No. Observations:                   12   AIC:                             71.27
Df Residuals:                       10   BIC:                             72.24
Df Model:                            1                                         
Covariance Type:             nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     58.8578      5.094     11.55

C:\ProgramData\Anaconda3\lib\site-packages\scipy\stats\stats.py:1603: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=12
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


Hệ số tương quan mạnh: 0.9988798361591897 ~ 1
Mô hình hồi quy: y(Tieu_thu_nang_luong) = 58.8578 + 4.8811 *  x(Dien_tich)

### b) Dự đoán mức tiêu thụ năng lượng hàng tháng của một căn hộ có diện tích 68 m². (1 điểm)

In [16]:
dien_tich = 68
print(f"Dự đoán mức tiêu thụ năng lượng hàng tháng của một căn hộ có diện tích 68 m²: {model.predict(exog=dict(Dien_tich=[dien_tich]))[0]:.2f} m²")

Dự đoán mức tiêu thụ năng lượng hàng tháng của một căn hộ có diện tích 68 m²: 390.77 m²
